scripts/compareBytecodes.ts

```diff
@@ -0,0 +1,228 @@
+import { execSync } from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+
+interface ComparisonResult {
+  contract: string
+  hardhatBytecode: string
+  foundryBytecode: string
+  match: boolean
+  hardhatLength: number
+  foundryLength: number
+  difference: string
+}
+
+function getHardhatArtifacts(): Map<string, string> {
+  const artifacts = new Map<string, string>()
+  const artifactsDir = path.join(__dirname, '..', 'build', 'contracts')
+
+  function scanDirectory(dir: string) {
+    const files = fs.readdirSync(dir)
+
+    for (const file of files) {
+      const fullPath = path.join(dir, file)
+      const stat = fs.statSync(fullPath)
+
+      if (stat.isDirectory()) {
+        scanDirectory(fullPath)
+      } else if (file.endsWith('.json')) {
+        const content = JSON.parse(fs.readFileSync(fullPath, 'utf8'))
+        if (content.bytecode && content.bytecode !== '0x') {
+          const contractName = file.replace('.json', '')
+          artifacts.set(contractName, content.bytecode)
```

Comment:
if we have two contracts with the same name but different paths, we could get a false positive / negative without this change

```suggestion
          if (artifacts.has(contractName)) {
            throw new Error(`Duplicate contract name found: ${contractName}`)
          }
          artifacts.set(contractName, content.bytecode)
```